saleor/dashboard/staff/forms.py

```diff
@@ -26,14 +26,33 @@ class Meta:
             'email': pgettext_lazy(
                 'Email', 'Email'),
             'is_active': pgettext_lazy(
-                'User active toggle', 'User is active')}
+                'User active toggle', 'User is active'),
+            'is_staff': pgettext_lazy(
+                'User staff toggle', 'User is staff')}
 
     def __init__(self, *args, **kwargs):
         self.user = kwargs.pop('user', None)
```

Comment:
Makes sense. But the user argument could still be passed as None, so we may still need to check user is not None in the permission checking logic.

Reply:
Passing `None` is just a mistake in code that would fail immediately since this is a `__init__` method. I don't think that `request.user` is ever `None` in Saleor, it's either AnonymousUser or just a User and this form is intended to be used by authenticated users. We could safely assume that `user` is a required argument and it won't be `None`.

Reply:
I agree with request.user should be safe. But I think None can be passed to ```user``` argument during form initialization such as ```StaffForm(request.POST or None, instance=staff_member, user=None)```?

Reply:
```
class Example:
    def __init__(self, *arg, **kwargs):
        pass


Example(somearg=None)
```

I think this code won't raise errors in Python?

Reply:
Of course, you could pass `None` here but we would you do that? From what I see all `StaffForm` occurrences could use `request.user`. We don't have to check that all arguments that a function takes are non-null if you expect them to be always non-null. E.g. we have the `create_order(cart, **something)` function which expects cart instance. We don't check if you accidentally passed `None` because that's would be your mistake and the code would fail immediately.

Reply:
I got what you mean. So we just assume correct types of argument are passed without checking it. It's Python's limitation comparing with other languages with type checking. The business logic shouldn't check the argument types.